cabal-testsuite/PackageTests/Regression/T5782Diamond/cabal.test.hs

```diff
@@ -0,0 +1,34 @@
+-- When Module.f is changed, with cabal <= 3.2 this non-deterministically fails
+-- to compile and, if it doesn't fail, it also non-deterministically gives
+-- a wrong answer (ignoring the change to Module.f in the output, despite
+-- recompiling, so probably the wrong library is linked in); when running
+-- manually on my machine, three changes to Module.hs are enough to trigger
+-- the error, often two are enough, even with cabal 3.2, even to get
+-- compilation error
+--   "Ambiguous module name `Module': it was found in multiple packages: issue5782-0.1 issue5782-0.1"
+-- not only the wrong result from exe run.
+--
+-- The dummy "--installdir=." is needed for cabal <= 3.2
+-- and also to match cabal output on different OSes
+-- (default installdir is different on various OSes).
+
+import Test.Cabal.Prelude
+main = cabalTest $
+    withSourceCopy . withDelay $ do
+        writeSourceFile "issue5782/src/Module.hs" "module Module where\nf = \"AAA\""
+        cabal "v2-install" ["--installdir=.", "--overwrite-policy=always", "issue5782"]
```

Comment:
I don't remember, does '`cabal-testsuite` specify config with non-user `store-dir`.

I think it should (and could specify `install-dir` which it normalizes as well when comparing outputs), so users' store is not polluted, neither affect runs.

Reply:
Do you mean something like https://github.com/haskell/cabal/blob/master/cabal-testsuite/PackageTests/Regression/T5409/use-different-versions-of-dependency-for-library-and-build-tool.test.hs#L18

Reply:
@Mikolaj that looks better though is still ugly.

`cabal-testsuite` is unfortunately not very good at isolating test environment (I mostly run it in docker container, see `Makefile`), which is not very nice.

Reply:
Thank you for the tips.  `withShorterPathForNewBuildStore` and Co let me avoid some path mismatches, etc., in the output, but MacOS still insists on processing internal libraries in a different order and Windows additionally still can't recognize that paths match. So I ended up adding `recordMode DoNotRecord`, which is fine in this case, because the problem manifests either as failed compilation or wrong exe output, which I do check.